source/structured_data.cc

```diff
@@ -783,13 +782,20 @@ namespace aspect
     void
     AsciiDataBoundary<dim>::update ()
     {
-      if (time_dependent && (this->get_time() - first_data_file_model_time >= 0.0))
+      // always initialize with the start time during model setup, even
+      // when restarting (to have identical setup in both cases)
+      double model_time = this->get_parameters().start_time;
+
+      // if we are past initialization use the current time instead
+      if (this->simulator_is_past_initialization())
+        model_time = this->get_time();
+
+      if (time_dependent == true)
```

Comment:
I don't understand why you removed this. Wouldn't this now try to load files even after the last available file is loaded and crash?

Reply:
I had another look at this PR, but @jdannberg I dont understand your comment (maybe it was placed on a different line by github in the meantime). Specifically, I am not sure what you mean by `why you removed this`, I dont see anything I removed in this code that would cause it to load another file after the last available file. After the last available file is loaded and the code can not find the next file it will set `time_dependent` to `false`, and that should prevent it from entering the if-condition again in the future.

Let me know if I misunderstood you, otherwise I think this is ready.